bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -365,95 +363,91 @@ private ICoreWebView2_13 initializeWebView_13(ICoreWebView2 webView) {
-		if(waitForPendingWebviewTasksToFinish) {
-			waitForFutureToFinish(lastWebViewTask);
-		}
-		return webViewWrapperFuture.join().webView_2;
+		return getWebViewWrapper(waitForPendingWebviewTasksToFinish).webView_2;
 	}
 
 	boolean isWebView_2Available() {
-		waitForFutureToFinish(webViewWrapperFuture);
-		return webViewWrapperFuture.join().webView_2 != null;
+		return getWebViewWrapper().webView_2 != null;
 	}
 
 	ICoreWebView2_10 getWebView_10(boolean waitForPendingWebviewTasksToFinish) {
-		if(waitForPendingWebviewTasksToFinish) {
-			waitForFutureToFinish(lastWebViewTask);
-		}
-		return webViewWrapperFuture.join().webView_10;
+		return getWebViewWrapper(waitForPendingWebviewTasksToFinish).webView_10;
 	}
 
 	boolean isWebView_10Available() {
-		waitForFutureToFinish(webViewWrapperFuture);
-		return webViewWrapperFuture.join().webView_10 != null;
+		return getWebViewWrapper().webView_10 != null;
 	}
 
 	ICoreWebView2_11 getWebView_11(boolean waitForPendingWebviewTasksToFinish) {
-		if(waitForPendingWebviewTasksToFinish) {
-			waitForFutureToFinish(lastWebViewTask);
-		}
-		return webViewWrapperFuture.join().webView_11;
+		return getWebViewWrapper(waitForPendingWebviewTasksToFinish).webView_11;
 	}
 
 	boolean isWebView_11Available() {
-		waitForFutureToFinish(webViewWrapperFuture);
-		return webViewWrapperFuture.join().webView_11 != null;
+		return getWebViewWrapper().webView_11 != null;
 	}
 
 	ICoreWebView2_12 getWebView_12(boolean waitForPendingWebviewTasksToFinish) {
-		if(waitForPendingWebviewTasksToFinish) {
-			waitForFutureToFinish(lastWebViewTask);
-		}
-		return webViewWrapperFuture.join().webView_12;
+		return getWebViewWrapper(waitForPendingWebviewTasksToFinish).webView_12;
 	}
 
 	boolean isWebView_12Available() {
-		waitForFutureToFinish(webViewWrapperFuture);
-		return webViewWrapperFuture.join().webView_12 != null;
+		return getWebViewWrapper().webView_12 != null;
 	}
 
 	ICoreWebView2_13 getWebView_13(boolean waitForPendingWebviewTasksToFinish) {
-		if(waitForPendingWebviewTasksToFinish) {
-			waitForFutureToFinish(lastWebViewTask);
-		}
-		return webViewWrapperFuture.join().webView_13;
+		return getWebViewWrapper(waitForPendingWebviewTasksToFinish).webView_13;
 	}
 
 	boolean isWebView_13Available() {
-		waitForFutureToFinish(webViewWrapperFuture);
-		return webViewWrapperFuture.join().webView_13 != null;
+		return getWebViewWrapper().webView_13 != null;
 	}
 
 	/*
 	 * Schedule a given runnable in a queue to execute when the webView is free and
 	 * has finished all the pending tasks queued before it.
 	 */
 	void scheduleWebViewTask(Runnable action) {
-		lastWebViewTask = lastWebViewTask.thenRun(() -> {
-			action.run();
+		lastWebViewTask = wakeDisplayAfterFuture(lastWebViewTask.thenRun(action::run));
+	}
+
+	<T> CompletableFuture<T> wakeDisplayAfterFuture(CompletableFuture<T> future) {
+		return future.handle((nil1, nil2) -> {
+			Display display = Display.getDefault();
```

Comment:
At best use the display of the browser here as Windows can have multiple displays

Reply:
Indeed, I adapted that.